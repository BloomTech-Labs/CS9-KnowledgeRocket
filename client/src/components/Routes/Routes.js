import React from "react";
import { BrowserRouter, Route } from "react-router-dom";
import Home from "../Home/Home";
import NavBar from "../NavBar/NavBar";
import ControlPanel from "../ControlPanel/ControlPanel";
import Auth from "../Auth/Auth";
import Rocket from "../Rocket/Rocket";
import Billing from "../Billing/Billing";
import Cohort from "../Cohort/Cohort";
import Settings from "../Settings/Settings";

export default () => (
    <BrowserRouter>
        <div className="routeContainer">
            <Route path="/rocket" component={NavBar} />
            <div className="Content_container">
                <Route path="/rocket" component={ControlPanel} />
                <Route path="/rocket" exact component={Rocket} />
                <Route path="/rocket/billing" exact component={Billing} />
                <Route path="/rocket/classes" exact component={Cohort} />
                <Route path="/rocket/settings" exact component={Settings} />
                <Route path="/rocket/auth" exact component={Auth} />
            </div>
            <Route path="/" exact component={Home} />
        </div>
    </BrowserRouter>
);
