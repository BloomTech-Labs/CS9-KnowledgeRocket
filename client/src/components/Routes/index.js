import React from 'react';
import { BrowserRouter, Route } from 'react-router-dom';
// Components
import Home from '../Home/Home';
import NavBar from '../NavBar/NavBar';
import ControlPanel from '../ControlPanel/ControlPanel';
import Auth from '../Auth/Auth';
import Rocket from '../Rocket/Rocket';
import RocketList from '../RocketList/RocketList';
import Billing from '../Billing/Billing';
import CohortList from '../CohortList/CohortList';
import Settings from '../Settings/Settings';

export default () => (
    <BrowserRouter>
        <div className="routeContainer">
            <Route path="/rocket" component={NavBar} />
            <div className="Content_container">
                <Route path="/rocket" component={ControlPanel} />
                <Route path="/rocket" exact component={RocketList} />
                <Route path="/rocket/single/:id" exact component={Rocket} />
                <Route path="/rocket/billing" exact component={Billing} />
                <Route path="/rocket/classes" exact component={CohortList} />
                <Route path="/rocket/settings" exact component={Settings} />
                <Route path="/rocket/auth" exact component={Auth} />
            </div>
            <Route path="/" exact component={Home} />
        </div>
    </BrowserRouter>
);
