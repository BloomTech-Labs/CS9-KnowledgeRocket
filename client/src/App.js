import React, { Component } from "react";
import Routes from "./components/Routes/";

class App extends Component {
  render() {
    return (
      // Only Routes Here, All Components get added inside the Routes File
      // Do not add components directly to app unless you do not wish them to be routed
      // A good example would be a "Permanent Nav Bar", however it is more functional if routed.
      <div>
        {/* A Nav Bar Could Go here depending on the architecture of the app */}
        <Routes />
        {/* A Footer Could Go here depending on the architecture of the app */}
      </div>
    );
  }
}

export default App;
