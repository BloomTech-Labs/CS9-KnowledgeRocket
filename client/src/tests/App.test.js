import React from 'react';
import ReactDOM from 'react-dom';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
// Components
import App from '../App';
import Routes from '../components/Routes';

Enzyme.configure({ adapter: new Adapter() });

describe('<App />', () => {
	it('renders without crashing', () => {
		const tree = renderer.create(<App />);
		expect(tree).toMatchSnapshot();
	});

	it('should render a <Routes /> component', () => {
		const wrapper = shallow(<App />);
		expect(wrapper.find(Routes));
	});
});
