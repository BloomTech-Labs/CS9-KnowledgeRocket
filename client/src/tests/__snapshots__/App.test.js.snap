// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders without crashing 1`] = `
<div>
  <div
    className="routeContainer"
  >
    <div
      className="Content_container"
    />
    <div
      className="sc-ifAKCX iIzwFJ"
    >
      <div
        className="sc-bdVaJa hFzOOv"
      >
        <div
          className="sc-bwzfXH jBrGUa"
        >
          <div>
            <img
              className="sc-bxivhb jiREkS"
              onClick={[Function]}
              src="/img/Logo_BW_SML.svg"
            />
          </div>
          <div>
            <button
              className="MuiButtonBase-root-27 MuiButton-root-1 MuiButton-text-3 MuiButton-flat-6 sc-htpNat dMcqFU"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex="0"
              type="button"
            >
              <span
                className="MuiButton-label-2"
              >
                Sign Up
              </span>
              <span
                className="MuiTouchRipple-root-30"
              />
            </button>
            <button
              className="MuiButtonBase-root-27 MuiButton-root-1 MuiButton-text-3 MuiButton-flat-6 sc-htpNat dMcqFU"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex="0"
              type="button"
            >
              <span
                className="MuiButton-label-2"
              >
                Sign In
              </span>
              <span
                className="MuiTouchRipple-root-30"
              />
            </button>
          </div>
        </div>
      </div>
      <div
        className="sc-EHOje iKFflp"
      >
        <h2
          className="sc-bZQynM fCAbAG"
        >
          What is a Knowledge Rocket?
        </h2>
        <div
          className="sc-htoDjs bsUlVr"
        >
          <div
            className="sc-dnqmqq fwKaRr"
          >
            <h3
              className="sc-gzVnrw caNqmJ"
            >
              Implementation
            </h3>
            <div
              className="sc-iwsKbI klDIwE"
            >
              <img
                className="sc-gZMcBi flLXQc"
                src="/img/amanda-sandlin-10508-unsplash.jpg"
              />
              <p
                className="sc-gqjmRU faHvtT"
              >
                Text for mid section here should not be very length but should contain enough information about this section.Text for mid section here should not be very length but should contain enough information about this section.
              </p>
            </div>
          </div>
          <div
            className="sc-dnqmqq fwKaRr"
          >
            <h3
              className="sc-gzVnrw caNqmJ"
            >
              Practice
            </h3>
            <div
              className="sc-iwsKbI klDIwE"
            >
              <img
                className="sc-gZMcBi cKudgZ"
                src="/img/erik-nielsen-783260-unsplash.jpg"
              />
              <p
                className="sc-gqjmRU faHvtT"
              >
                Text for mid section here should not be very length but should contain enough information about this section.Text for mid section here should not be very length but should contain enough information about this section.Text for mid section here should not be very length but should contain enough information about this section.
              </p>
            </div>
          </div>
          <div
            className="sc-dnqmqq fwKaRr"
          >
            <h3
              className="sc-gzVnrw caNqmJ"
            >
              Results
            </h3>
            <div
              className="sc-iwsKbI klDIwE"
            >
              <img
                className="sc-gZMcBi cMjvs"
                src="/img/bryan-minear-315814-unsplash.jpg"
              />
              <p
                className="sc-gqjmRU faHvtT"
              >
                Text for mid section here should not be very length but should contain enough information about this section.Text for mid section here should not be very length but should contain enough information about this section.
              </p>
            </div>
          </div>
        </div>
      </div>
      <div
        className="sc-VigVT dFrDii"
      >
        <button
          className="MuiButtonBase-root-27 MuiButton-root-1 MuiButton-text-3 MuiButton-flat-6 sc-htpNat dMcqFU"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex="0"
          type="button"
        >
          <span
            className="MuiButton-label-2"
          >
            CONTACT US
          </span>
          <span
            className="MuiTouchRipple-root-30"
          />
        </button>
        <button
          className="MuiButtonBase-root-27 MuiButton-root-1 MuiButton-text-3 MuiButton-flat-6 sc-htpNat dMcqFU"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex="0"
          type="button"
        >
          <span
            className="MuiButton-label-2"
          >
            SIGN UP
          </span>
          <span
            className="MuiTouchRipple-root-30"
          />
        </button>
        <button
          className="MuiButtonBase-root-27 MuiButton-root-1 MuiButton-text-3 MuiButton-flat-6 sc-htpNat dMcqFU"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex="0"
          type="button"
        >
          <span
            className="MuiButton-label-2"
          >
            SIGN IN
          </span>
          <span
            className="MuiTouchRipple-root-30"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;
